ingest/expression_files/expression_files.py

```diff
@@ -150,9 +150,15 @@ def get_cell_names_from_study_file_id(study_id, study_file_id, client):
         )
         # The query did not return results
         if not query_results:
-            raise ValueError(
-                "There are study files that do not have cell names associated with them"
-            )
+            # check if the other expression matrices are "parsing", in which case ignore
+            for matrix_file_id in study_files_ids:
+                matrix_file = list(client['study_files'].find({"_id": matrix_file_id})).pop()
+                if matrix_file['parse_status'] == 'parsing':
+                    continue
+                else:
+                    raise ValueError(
```

Comment:
now that we're adding the 'one parse at a time' restriction, do we still want to raise this error?  It seems like ingest pipeline shouldn't be in the business of throwing errors based on previously ingested files.  If we are forbidding expression matrices with zero cells, that check should happen at ingest time

Reply:
I don't have any objection to that except that if we remove it, we should add a validation somewhere else that would throw an error if we don't have any cell names to store (which is what this is checking in reverse).  The only reason that we're instituting the "one parse at a time" rule for expression matrices of the same type is to ensure that we can validate cells across all matrices, and concurrent parses cannot do this.

Reply:
2bb3ba5